cluster/terraform/main.tf

```diff
@@ -61,6 +62,21 @@ resource "azurerm_kubernetes_cluster" "main" {
   lifecycle { ignore_changes = [tags] }
 }
 
+resource "azurerm_kubernetes_cluster_node_pool" "node_pools" {
+  for_each = var.node_pools
+
+  name                  = each.key
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.main.id
+  vm_size               = "Standard_D2_v2"
+  enable_auto_scaling   = true
+  min_count             = each.value.min_count
+  max_count             = each.value.max_count
+  node_labels           = {"node_pool": "${each.key}"}
+  node_taints           = ["node_pool=${each.key}:NoSchedule"]
```

Comment:
Can we remove node_taints and simply use node_selector on the deployment instead of tolerations?

Reply:
Hey @saliceti, I've removed both `node_labels` and `node_taints` as it seems you can select node pools based on the name of node pools - https://github.com/DFE-Digital/apply-for-teacher-training/blob/2fd350d9aba88283fb18225da986a8671252602d/terraform/modules/kubernetes/main.tf#L21

Would you like the `node_pool` label being put back on or leaving off? 
AKS labels: https://learn.microsoft.com/en-us/azure/aks/use-labels#reserved-system-labels

Reply:
That's even better, thanks